### DISCUSSION TITLE
List of IP addresses that match ota.tasmota.com

### DISCUSSION THREAD

**Discussion by GijsTimmers** [arendst/Tasmota#21007] [Q&A] [answered]
Is there a known list of possible IP addresses that `ota.tasmota.com` uses?

I'm asking this because I want to secure my network by limiting the IPs that can be reached.

**Comment by s-hadinger** [comment, marked as answer]
`ota.tasmota.com` used Amazon Cloudfront as a CDN. There are many (really many) IP addresses.

**Reply by sfromis** [reply]
And nothing would stop the same IP addresses also being used for sites you'd not want to use. With modern networking, where IPv4 addresses are a scarce commodity, it is not workable to blacklist (or whitelist) specific IP addresses, except for whole blocks where the ownership is sufficient criteria.

**Reply by s-hadinger** [reply]
If you're really interested, you will find all the ranges here : https://docs.aws.amazon.com/vpc/latest/userguide/aws-ip-ranges.html

**Reply by GijsTimmers** [reply]
Hmm... You're right @sfromis . Haven't thought of it this way.

So I was thinking, there's no proper way of securing my network totally:
- I could whitelist Tasmota OTA server IPs, but a hacker could serve a malicious website on any of those IPs too;
- Blocking all traffic except on port 123 for the NTP server, as described [here](https://tasmota.github.io/docs/Securing-your-IoT-from-hacking/) works, but there's nothing stopping a hacker to serve a malicious website on that port

**Reply by sfromis** [reply]
If playing security theater is your idea of having fun, you can set up in-house NTP and OTA servers, meaning that Tasmota will usually not need to reach any external sites.

I'd say that the risk with OTA server access is on the low end, as your device would already have to be hacked before becoming the source of bad traffic. Or Amazon would have to be hacked for an attacker to take over the domain.

Similar considerations for NTP, except that this is low value to a cracker, as it could only be to provide a bad time, or (in principle) to exploit a known vulnerability in the NTP client code.

**Reply by barbudor** [reply]
Just don't give your devices access to the internet and serve the files locally
A local mirror of Ota.tasmota.com can fetch the binaries over hhtps which is safe and serve the files over http internally
You can even recompile from the sources in case ota.tasmota.com has been compromised 😱

Same for NTP. Most router's can act as NTP servers. No need to go outside.

**Reply by GijsTimmers** [reply]
Right. Thanks @sfromis @barbudor .

**Reply by sfromis** [reply]
On ESP32-based devices, you can also use https for the OTA server, while ESP8266-based devices can only be http. In my setup, both NTP and caching OTA servers are my own, but I do not worry about blocking Internet access from Tasmota.
